### ISSUE TITLE
Multi-stage piecewise polytrope

### ISSUE THREAD

**Issue by nilsvu** [sxs-collaboration/spectre#5512]
Our `PiecewisePolytropicFluid` equation of state currently has only 2 stages. It would be useful to generalize this to an arbitrary number of stages.

**Comment by nilsdeppe** [comment]
Yes, but I'm not convinced one should use piecewise polytropes in real simulations. The enthalpy parametrization seems to be superior in every way possible for MHD simulations
